src/modules/Game.class.js

```diff
@@ -21,27 +28,108 @@ class Game {
    * initial state.
    */
   constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+    if (!initialState) {
+      this.#state = new Array(4).fill(0);
+
+      for (const i in this.#state) {
+        this.#state[i] = new Array(4).fill(0);
+      }
+
+      this.#initialState = this.#getStateCopy();
+    } else {
+      this.#state = this.#getStateCopy(initialState);
+      this.#initialState = this.#getStateCopy();
+    }
+    this.#SIZE = 4;
+    this.#status = 'idle';
+  }
+
+
+  // This is a required method
+  moveLeft() {
+    if (this.#status === 'playing') {
+      const previousState = this.#getStateCopy().flat();
+
+      this.#makeMove();
+
+      if (this.#state.flat().some((e, i) => e !== previousState[i])) {
+        this.#randomizer();
+        this.#checkIfLose();
+      }
```

Comment:
I think that is not possible, because each movement make some operations( reverse, tranpose etc. ).
For example the left movement just call makeMove() method but the right movement use the sequence of methods reverse()-> makeMove() -> reverse(). Each movement have your own sequence of methods always reusing code in this methods(reverse, makeMove, tranpose);